src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,6 +171,11 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
+        public static void RequiresCanRead(Expression expression, string paramName)
+        {
+            RequiresCanRead(expression, paramName, -1);
+        }
```

Comment:
I think this is a reasonable balance; it's worth having the overload for the slight size decrease it produces considering that it's called a great many times, but not worth having a full duplication within it.